tools/traffic/table/blob_table.go

```diff
@@ -0,0 +1,125 @@
+package table
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+)
+
+// BlobTable tracks blobs written by the traffic generator. This is a thread safe data structure.
+type BlobTable struct {
+
+	// blobs contains all blobs currently tracked by the requiredReads.
+	blobs []*BlobMetadata
+
+	// size describes the total number of blobs currently tracked by the requiredReads.
+	// size may be smaller than the capacity of the blobs slice.
+	size uint
+
+	// lock is used to synchronize access to the requiredReads.
+	lock sync.Mutex
+}
+
+// NewBlobTable creates a new BlobTable instance.
+func NewBlobTable() BlobTable {
+	return BlobTable{
+		blobs: make([]*BlobMetadata, 1024),
```

Comment:
why don't we make the size of this slice variable?

Reply:
That would work as long as we only add blobs. When blobs are removed from the table, the size of the table may actually be smaller than the size of this slice.